src/bindgen/ir/enumeration.rs

```diff
@@ -448,6 +458,71 @@ impl Source for Enum {
                 out.close_brace(true);
             }
 
+
+            // Emit convenience methods
+            if config.language == Language::Cxx &&
+               config.enumeration.derive_helper_methods(&self.annotations) {
+
+                for variant in &self.variants {
+                    out.new_line();
+                    out.new_line();
+
+                    let arg_renamer = |name: &str| {
+                        config
+                            .function
+                            .rename_args
+                            .as_ref()
+                            .unwrap_or(&RenameRule::GeckoCase)
+                            .apply_to_snake_case(name, IdentifierType::FunctionArg)
+                    };
+
+                    write!(out, "static {} {}(", self.name, variant.name);
+
+                    let skip_fields = if separate_tag { 0 } else { 1 };
+
+                    if let Some((_, ref body)) = variant.body {
+                        out.write_vertical_source_list(
+                            &body
+                                .fields.iter().skip(skip_fields)
+                                .map(|&(ref name, ref ty, _)| {
+                                    // const-ref args to constructor
+                                    (format!("const& {}", arg_renamer(name)), ty.clone())
+                                })
+                                .collect(),
+                            ListType::Join(","),
+                        );
+                    }
+
+                    write!(out, ")");
+                    out.open_brace();
+
+                    write!(out, "{} result;", self.name);
+
+                    if let Some((ref variant_name, ref body)) = variant.body {
+                        for &(ref field_name, ..) in body.fields.iter().skip(skip_fields) {
+                            out.new_line();
+                            write!(out, "result.{}.{} = {};",
+                                variant_name, field_name, arg_renamer(field_name));
+                        }
+                    }
+
+                    out.new_line();
+                    write!(out, "result.tag = {}::{};", enum_name, variant.name);
+                    out.new_line();
+                    write!(out, "return result;");
+                    out.close_brace(false);
+                }
+
+                for variant in &self.variants {
+                    out.new_line();
+                    out.new_line();
+                    write!(out, "bool Is{}()", variant.name);
```

Comment:
Could we have `Is{variant.name}` be transformed by the rename rules for functions?